docs/internals/contributing/writing-code/coding-style.txt

```diff
@@ -56,6 +48,44 @@ Python style
           """
           pass
 
+Imports
+-------
+
+* Use convenience imports whenever available. For example, do this::
+
+      from django.views.generic import View
+
+  Don't do this::
+
+      from django.views.generic.base import View
+
+* Put imports in five groups: future, standard library, third-party libraries,
+  other Django components, the same component; sort lines in each group
+  alphabetically by the full module name. Place ``import module`` statements
+  after ``from module import objects``.
+
+* On each line, list imported modules first, then classes, and then other
+  objects also ordering each group alphabetically, with upper case before
+  lower.
+
+* Break long lines using parentheses, indenting continuation lines by 4 spaces.
+  For example::
+
+        from __future__ import unicode_literals
+
+        import email.message
+        import turtle
+
+        import bcrypt
+
+        from django.contrib.messages import MessageFailure, ERROR, error
+        from django.db import (models, transaction, DataError, DatabaseError, Error, IntegrityError,
```

Comment:
I'd leave the closing brace on the 3rd line.

Ending the first line after the opening parenthesis is currently more or less as common as starting to list objects right away (173 cases compared to 152, but 88 files versus 124. Making a line break before the closing parenthesis isn't that popular however, with 60 files versus 148.

I can't guarantee the full accuracy of these stats (if any at all ;-), some of the commands used to get the numbers:
* ``grep -RPzl 'from\s+[\w.]+\s+import\s+\(' --include=*.py . | wc -l`` (205 files in total)
* ``grep -RPzl 'from\s+[\w.]+\s+import\s+\(\s*(\n|#)' --include=*.py . | wc -l`` (88)
* ``grep -RPzl 'from\s+[\w.]+\s+import\s+\(\w' --include=*.py . | wc -l`` (124)
* ``grep -RPzl 'from\s+[\w.]+\s+import\s+\([^\)]*\n\)' --include=*.py . | wc -l`` (60)
* ``grep -RPzl 'from\s+[\w.]+\s+import\s+\([^\)]*\w\)' --include=*.py . | wc -l`` (148)

What do you think?